modules/nextflow/src/main/groovy/nextflow/file/FilePorter.groovy

```diff
@@ -331,7 +331,7 @@ class FilePorter {
                     // remove the target file that could be have partially downloaded
                     cleanup(stagePath)
                     // check if a stage/download retry is allowed
-                    if( count++ < maxRetries && e !instanceof NoSuchFileException && e !instanceof InterruptedIOException && !Thread.currentThread().isInterrupted() ) {
+                    if( count++ < maxRetries && e !instanceof NoSuchFileException && (e instanceof SocketTimeoutException || e !instanceof InterruptedIOException) && !Thread.currentThread().isInterrupted() ) {
```

Comment:
Let's move ` e !instanceof NoSuchFileException && (e instanceof SocketTimeoutException || e !instanceof InterruptedIOException)` into a separate method e.g. `recoverableError(e)` otherwise it's impossible to understand it. 

I would also add `SocketException` at this point

Reply:
Changed in last commit